great_expectations/metrics/domain.py

```diff
@@ -58,3 +59,16 @@ class ColumnValues(Values):
     """
 
     column: NonEmptyString
+
+
+class Batch(Domain):
+    """A domain type for metrics that compute over an entire batch.
+
+    The Batch domain type is used to define metrics that evaluate conditions or compute
+    values for an entire table. This class is intended to be used as a mixin
+    with the Metric class when defining a new Metric.
+    """
+
+    table: Optional[NonEmptyString] = None
```

Comment:
When is `table` `None`? When computing over a dataframe what is `table`?

Reply:
i know this is weird, but i think the better question is, when is table _not_ `None`? I tested against a couple backends, and it appears to be unused. It's also causing the huge amount of logs reported in `gx-398`. I'm working on removing it altogether in https://github.com/great-expectations/great_expectations/pull/10951.

Reply:
I actually had to do this same thing in my PR due to testing with pandas. If we can remove it altogether that would be great! I'm not sure that we can though because I haven't looked at all of the legacy metric implementations.